File: frontend/src/services/tenantService.js
import api from "./api";

const tenantService = {

    assignTenant: async (applicationId, data) => {
        const response = await api.post(
            `/tenants/applications/${applicationId}/assign`,
            data
        );
        return response.data;
    },

    getTenantsByProperty: async (propertyId) => {
    const response = await api.get(
        `/tenants/property/${propertyId}`
    );

    return response.data;
},

vacateTenant: async (tenantId) => {
    const response = await api.patch(
        `/tenants/${tenantId}/vacate`
    );

    return response.data;
},

getMyPg: async () => {

    const response = await api.get(
        "/tenants/my-pg"
    );

    return response.data;
},

};

export default tenantService;

File: backend/src/main/java/com/nishtha/RoomEase/tenant/service/TenantService.java
package com.nishtha.RoomEase.tenant.service;

import com.nishtha.RoomEase.tenant.dto.AssignTenantRequest;
import com.nishtha.RoomEase.tenant.dto.TenantPGResponse;
import com.nishtha.RoomEase.tenant.dto.TenantResponse;

import java.util.List;

public interface TenantService {

    TenantResponse assignTenant(Long applicationId, AssignTenantRequest request);

    List<TenantResponse> getTenantsByProperty(Long propertyId);

    TenantResponse vacateTenant(Long tenantId);

    TenantPGResponse getMyPg();

}